hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -540,48 +543,37 @@ private static Object transformEnumValue(final ComparisonNode node, final String
-            }
 
-            addOperatorPredicate(node, getMapValueFieldPath(enumField, fieldPath), transformedValues, transformedValue,
-                    value, singleList, database);
-            return Collections.unmodifiableList(singleList);
+            final Predicate valuePredicate = addOperatorPredicate(node, getMapValueFieldPath(enumField, fieldPath),
+                    transformedValues, transformedValue, value, database);
+
+            return toSingleList(mapPredicate != null ? cb.and(mapPredicate, valuePredicate) : valuePredicate);
         }
 
-        private void addOperatorPredicate(final ComparisonNode node, final Path<Object> fieldPath,
+        private Predicate addOperatorPredicate(final ComparisonNode node, final Path<Object> fieldPath,
                 final List<Object> transformedValues, final Object transformedValue, final String value,
-                final List<Predicate> singleList, final Database database) {
+                final Database database) {
             switch (node.getOperator().getSymbol()) {
             case "==":
-                singleList.add(getEqualToPredicate(transformedValue, fieldPath, database));
-                break;
+                return getEqualToPredicate(transformedValue, fieldPath, database);
             case "!=":
-                singleList.add(getNotEqualToPredicate(transformedValue, fieldPath, database));
-                break;
+                return getNotEqualToPredicate(transformedValue, fieldPath, database);
             case "=gt=":
-                singleList.add(cb.greaterThan(pathOfString(fieldPath), value));
-                break;
+                return cb.greaterThan(pathOfString(fieldPath), value);
             case "=ge=":
-                singleList.add(cb.greaterThanOrEqualTo(pathOfString(fieldPath), value));
-                break;
+                return cb.greaterThanOrEqualTo(pathOfString(fieldPath), value);
             case "=lt=":
-                singleList.add(cb.lessThan(pathOfString(fieldPath), value));
-                break;
+                return cb.lessThan(pathOfString(fieldPath), value);
             case "=le=":
-                singleList.add(cb.lessThanOrEqualTo(pathOfString(fieldPath), value));
-                break;
+                return cb.lessThanOrEqualTo(pathOfString(fieldPath), value);
             case "=in=":
-                singleList.add(getInPredicate(transformedValues, fieldPath));
-                break;
+                return getInPredicate(transformedValues, fieldPath);
             case "=out=":
-                singleList.add(getOutPredicate(transformedValues, fieldPath));
-                break;
+                return getOutPredicate(transformedValues, fieldPath);
             default:
-                LOGGER.info("operator symbol {} is either not supported or not implemented");
+                throw new RSQLParameterSyntaxException("operator symbol {" + node.getOperator().getSymbol()
```

Comment:
Instead of passing `null` into constructer here, you could also introduce a new one without `cause` parameter, since `AbstractServerRtException` is already offering one:
```
public RSQLParameterSyntaxException(final String message) {
        super(message, SpServerError.SP_REST_RSQL_SEARCH_PARAM_SYNTAX);
}
```

Reply:
fixed